test/close-pipelining.test.js

```diff
@@ -3,43 +3,40 @@
 const t = require('tap')
 const test = t.test
 const Fastify = require('..')
-const autocannon = require('autocannon')
-
-// this tests on windows takes an unusually large amount of time.
-// https://github.com/fastify/fastify/issues/2470
-t.setTimeout(45000)
+const { Client } = require('undici')
 
 test('Should return 503 while closing - pipelining', t => {
-  const fastify = Fastify()
+  const fastify = Fastify({
+    return503OnClosing: true
+  })
 
   fastify.get('/', (req, reply) => {
     fastify.close()
     reply.send({ hello: 'world' })
   })
 
-  fastify.listen(0, err => {
+  fastify.listen(0, async err => {
     t.error(err)
 
-    const instance = autocannon({
-      url: 'http://localhost:' + fastify.server.address().port,
-      pipelining: 1,
-      connections: 1,
-      amount: 10
+    const instance = new Client('http://localhost:' + fastify.server.address().port, {
+      pipelining: 1
     })
 
     const codes = [200, 503]
-    instance.on('response', (client, statusCode) => {
-      t.strictEqual(statusCode, codes.shift())
-    })
-
-    instance.on('done', () => {
+    // eslint-disable-next-line no-unused-vars
+    for (const _ of Array(codes.length)) {
```

Comment:
Can it be written as?
```js
    for (const code of Array(codes.length)) {
      const data = await instance.request(
        { path: '/', method: 'GET' }
      )
	 t.strictEqual(data.statusCode, code)
    }
```

Reply:
fair point. Just this will be deleted:
```js
      t.strictEqual(codes.length, 0)
```
Because the code array is not mutated anymore

Reply:
Fine to me the loop will ensure that all cases are processed.